### PR TITLE
Track file offset for each matching log line, and output as result json

### PR DESCRIPTION
**Pull request by jspaleta**
Adjusting how the analyzer is implemented so that file offset for each matching line is tracked.

This is useful for when log file has non-unique matching log lines as you can use the byte offset information to seek to the specific matching line in the log file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -476,7 +476,9 @@ func processLogFile(file string, enc *json.Encoder) (int, error) {
 	offset := state.Offset
 	// Are we looking at freshly rotated file since last time we run?
 	// If so let's reset the offset back to 0 and read the file again
-
+	if offset < 0 {
+		offset = 0
+	}
```
Comment (echlebek) [inline main.go:481]: Under what circumstances does this condition occur? Seems surprising.

Comment (jspaleta) [reply main.go:481]: It shouldn't occur.  But if it does I want to catch it. Should I error here instead?

Comment (echlebek) [reply main.go:481]: Hmm I think so, it's basically an assertion failure, may as well panic.

Comment (jspaleta) [reply main.go:481]: okay ill change it as a non zero exit with a message

**Review by echlebek — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (jspaleta) [reply main.go]: latest commits change the condition to an error, and adds a test for the error condition

**Pull request merged**